### PR TITLE
Migrate IexecPoco2 unit tests related to kitty

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: test/byContract/IexecPoco/04_finalize.test.ts [hunk 1]
```diff
@@ -356,6 +356,135 @@ describe('IexecPoco2#finalize', async () => {
...
+            expectedSchedulerKittyPartReward: 999_999_999,
+        },
+    ].forEach((testArgs) => {
+        const {
+            testInfo,
+            workerpoolPriceToFillKitty,
+            expectedKittyBeforeFinalize,
+            expectedSchedulerKittyPartReward,
+        } = testArgs;
+        it(`Should finalize task with scheduler kitty part reward (${testInfo})`, async () => {
```
Comment (gfournieriExec) [inline test/byContract/IexecPoco/04_finalize.test.ts:385]:
```suggestion
        it(`Should finalize task with scheduler kitty part reward : ${testInfo} `, async () => {
```
More readable on logs, with the time spent on the test already in "()"

File: test/byContract/IexecPoco/04_finalize.test.ts [hunk 2]
```diff
@@ -356,6 +356,135 @@ describe('IexecPoco2#finalize', async () => {
...
+            // Fill kitty
+            const kittyAddress = await iexecPoco.kitty_address();
+            const kittyFillingDealVolume = 1;
+            const kittyFillingSchedulerDealStake = await iexecWrapper.computeSchedulerDealStake(
+                workerpoolPriceToFillKitty,
+                kittyFillingDealVolume,
+            );
+            const kittyFillingSchedulerTaskStake =
+                kittyFillingSchedulerDealStake / kittyFillingDealVolume;
+            const kittyFilling = await iexecWrapper.signAndMatchOrders(
```
Comment (gfournieriExec) [inline test/byContract/IexecPoco/04_finalize.test.ts:395]: I don't think `kittyFilling` "alone" is the right name, like the others that have kittyFillingX with X as the description of what is in the variable

Comment (gfournieriExec) [reply test/byContract/IexecPoco/04_finalize.test.ts:395]: At least here , I like it after with `kittyFilling.dealId` etc

File: test/byContract/IexecPoco/04_finalize.test.ts [hunk 3]
```diff
@@ -356,6 +356,135 @@ describe('IexecPoco2#finalize', async () => {
...
+                    volume: kittyFillingDealVolume,
+                    salt: ethers.utils.id(new Date().toISOString()),
+                }).toArray(),
+            );
+            await iexecPoco
+                .initialize(kittyFilling.dealId, kittyFilling.taskIndex)
+                .then((tx) => tx.wait());
+            const kittyFrozenBeforeClaim = (await iexecPoco.frozenOf(kittyAddress)).toNumber();
+            const task = await iexecPoco.viewTask(kittyFilling.taskId);
+            await time.setNextBlockTimestamp(task.finalDeadline);
```
Comment (gfournieriExec) [inline test/byContract/IexecPoco/04_finalize.test.ts:413]:
```suggestion
            await time.setNextBlockTimestamp(await iexecPoco.viewTask(kittyFilling.taskId).finalDeadline);
```
(?)

File: test/byContract/IexecPoco/04_finalize.test.ts [hunk 4]
```diff
@@ -356,6 +356,135 @@ describe('IexecPoco2#finalize', async () => {
...
+            await expect(iexecPoco.claim(kittyFilling.taskId))
+                .to.emit(iexecPoco, 'Transfer')
+                .withArgs(iexecPoco.address, kittyAddress, kittyFillingSchedulerTaskStake)
+                .to.emit(iexecPoco, 'Reward')
+                .withArgs(kittyAddress, kittyFillingSchedulerTaskStake, kittyFilling.taskId)
+                .to.emit(iexecPoco, 'Transfer')
+                .withArgs(kittyAddress, iexecPoco.address, kittyFillingSchedulerTaskStake)
+                .to.emit(iexecPoco, 'Lock')
+                .withArgs(kittyAddress, kittyFillingSchedulerTaskStake)
+                .to.changeEtherBalances([iexecPoco, kittyAddress], [0, 0]);
```
Comment (gfournieriExec) [inline test/byContract/IexecPoco/04_finalize.test.ts:423]: why ether balances and not changeTokenBalances ?

Comment (jeremyjams) [reply test/byContract/IexecPoco/04_finalize.test.ts]: Thanks a lot!

File: test/utils/IexecWrapper.ts [hunk 5]
```diff
@@ -104,7 +104,7 @@ export class IexecWrapper {
                 this.accounts.anyone,
             ).workerpool_stake_ratio()
         ).toNumber();
-        return ((workerpoolPrice * stakeRatio) / 100) * volume;
+        return Math.floor((workerpoolPrice * stakeRatio) / 100) * volume;
```
Comment (gfournieriExec) [inline test/utils/IexecWrapper.ts:107]: Why this change now ?

Comment (jeremyjams) [reply test/utils/IexecWrapper.ts:107]: We need to mimic what is done in the contract, workerpool price is not always divisible without remainder when applying 30% to get staking ratio

File: test/byContract/IexecPoco/04_finalize.test.ts [hunk 6]
```diff
@@ -356,6 +356,150 @@ describe('IexecPoco2#finalize', async () => {
...
+            } = testArgs;
+            it(`Should finalize task with scheduler kitty part reward where ${testInfo}`, async () => {
+                // Fill kitty
+                const kittyAddress = await iexecPoco.kitty_address();
+                const kittyFillingDealVolume = 1;
+                const kittyFillingSchedulerDealStake = await iexecWrapper.computeSchedulerDealStake(
+                    workerpoolPriceToFillKitty,
+                    kittyFillingDealVolume,
+                );
+                const kittyFillingSchedulerTaskStake =
```
Comment (gfournieriExec) [inline test/byContract/IexecPoco/04_finalize.test.ts:395]: we should math.floor this operation then ? since here it's volume = 1

Comment (jeremyjams) [reply test/byContract/IexecPoco/04_finalize.test.ts:395]: In this cas dealStake = taskStake so flooring not needed, do you want me to add it?

Comment (gfournieriExec) [reply test/byContract/IexecPoco/04_finalize.test.ts:395]: no : * volume / volume

**Review by gfournieriExec — APPROVED** [review]
Perfect thanks :)
